tmt/export.py

```diff
@@ -218,6 +253,68 @@ def bz_set_coverage(bz_instance, bug_ids, case_id):
     if not overall_pass:
         raise ConvertError("Failed to link the case to bugs.")
 
+    echo(style("Linked to Bugzilla: {}.".format(
+        " ".join([f"BZ#{bz_id}" for bz_id in bug_ids])), fg='magenta'))
+
+
+def _get_polarion_ids(query_result, preferred_project=None):
+    """ Return case and project ids from query results """
+    if not query_result:
+        return 'None', None
+    if len(query_result) == 1:
+        return query_result[0].work_item_id, query_result[0].project_id
+
+    if preferred_project:
+        try:
+            return next(
+                item.work_item_id for item in query_result
+                if item.project_id == preferred_project), preferred_project
+        except StopIteration:
+            pass
+
+    for result in query_result:
+        # If multiple cases are found prefer cases from other projects
+        # than these legacy ones
+        if str(result.project_id) not in LEGACY_POLARION_PROJECTS:
+            return result.work_item_id, result.project_id
+
+    return query_result[0].work_item_id, query_result[0].project_id
+
+
+def get_polarion_case(data, preferred_project=None):
+    """ Get Polarion case through couple different methods """
+    import_polarion()
+    polarion_id = 'None'
+    project_id = None
+
+    # Search by UUID
+    if data.get(ID_KEY):
+        query_result = PolarionWorkItem.query(
+            data.get(ID_KEY), fields=['work_item_id', 'project_id'])
+        polarion_id, project_id = _get_polarion_ids(query_result, preferred_project)
+    # Search by TCMS Case ID
```

Comment:
yeah, there is TCMS case ID field, so all cases synced from nitrate have this populated (done for most of baseos cases) and is the easiest way to find a case in polarion, unless there is a match for this really short number in any other field...then it might cause problems actually, that's why I added preferred project attribute

Reply:
yeah, there is TCMS case ID field, so all cases synced from nitrate have this populated (done for most of baseos cases) and is the easiest way to find a case in polarion, unless there is a match for this really short number in any other field...then it might cause problems actually, that's why I added preferred project attribute